### PR TITLE
User, Portfolio, Stocks, Leagues, and Transaction APIs work

### PR DESCRIPTION
**Pull request by rohithdara**

### CODE REVIEW

File: Scripts/setup.sh [hunk 1]
```diff
@@ -93,9 +93,9 @@ echo -e "${GREEN}Successfully setup service module${RESTORE}\n"
 
 ###############################################################################
 # Setup the StockDog database
-echo "Creating the StockDog database"
```
Comment (ndara) [inline Scripts/setup.sh:96]: why delete this line?

File: Service/src/app.py [hunk 2]
```diff
@@ -42,13 +42,14 @@ def setup():
...
 @app.route('/')
-@app.route('/api/v1.0')
+@app.route('/api')
 def index():
    return "What's good StockDog!"
 
 
 @app.errorhandler(404)
 def not_found(error):
+   print(error)
```
Comment (ndara) [inline Service/src/app.py]: no print statements in code

remove debugging statements, any statements that need to stay should use the logger object

File: Service/src/app.py [hunk 3]
```diff
@@ -42,13 +42,14 @@ def setup():
 
 
 @app.route('/')
-@app.route('/api/v1.0')
+@app.route('/api')
```
Comment (ndara) [inline Service/src/app.py:45]: why remove the versioning?

File: Service/src/routes/charts.py [hunk 4]
```diff
@@ -8,18 +8,21 @@
...
 from request_validator.schemas import charts_schema
 from util.error_map import errors
+from util.db_connection import getConfigFilePath
 
 DAY = '1d'
 MONTH = '1m'
 YEAR = '1y'
 
-IEX_DATETIME_FORMAT = '%Y%m%d %H:%M'
+IEX_DATETIME_FORMAT = '%Y-%m-%d %H:%M'
```
Comment (ndara) [inline Service/src/routes/charts.py:17]: I'm guessing IEX changed their format then?

File: Service/src/routes/charts.py [hunk 5]
```diff
@@ -8,18 +8,21 @@
...
 IEX_DATE_FORMAT = '%Y-%m-%d'
 DATETIME_FORMAT = '%Y-%m-%d %H:%M:%S'
 
 charts_api = Blueprint('charts_api', __name__)
 
-URL_PREFIX = 'https://api.iextrading.com/1.0/stock/'
+URL_PREFIX = 'https://cloud.iexapis.com/v1/stock/'
+
+CONFIG_FILE_PATH = "/Users/rohithdara/dev/StockDog/Service/src/routes/config.json"
```
Comment (ndara) [inline Service/src/routes/charts.py]: oh hell no lol, you can't reference your own "rohithdara" folder in this code

you need to have it generic such that I (or anyone else) can pull it down and have it work

This is a blocker for merging

File: Service/src/app.py (hunk 2 above)
Comment (ndara) [reply Service/src/app.py]: this is a blocker for merging

File: Service/src/routes/charts.py [hunk 6]
```diff
@@ -43,6 +52,7 @@ def get_history(ticker, length):
 
    try:
       response = rawResponse.json()
+      g.log.info(response, isPprint=True)
```
Comment (ndara) [inline Service/src/routes/charts.py]: this is a fat log that shouldn't be printed
blocker for merge

File: Service/src/routes/charts.py [hunk 7]
```diff
@@ -96,8 +106,13 @@ def formatDataInterDay(jsonData):
 
 
 def formatDataIntraday(jsonData):
-   data = [] 
+   data = []
+   NoneItemsIEX = 0
```
Comment (ndara) [inline Service/src/routes/charts.py]: poorly named variable -- you can do better naming here

with Python, there's no explicit types so you have to use variable names to communicate the type

If I understand this variable correctly it's `emptyIEXDataObjectCount`

File: Service/src/routes/charts.py [hunk 8]
```diff
@@ -96,8 +106,13 @@ def formatDataInterDay(jsonData):
 
 
 def formatDataIntraday(jsonData):
-   data = [] 
+   data = []
+   NoneItemsIEX = 0 
    for item in jsonData:
+      if item['average'] is None:
+         print(item)
```
Comment (ndara) [inline Service/src/routes/charts.py]: no print statement --> blocker for merge

use the logger object even when you're debugging

File: Service/src/routes/charts.py [hunk 9]
```diff
@@ -107,6 +122,8 @@ def formatDataIntraday(jsonData):
          'epochTime' : itemTime.timestamp(),
          'price' : item['average']
       })
+   # if NoneItemsIEX > 2:
```
Comment (ndara) [inline Service/src/routes/charts.py]: no commented code, this is a code smell, either remove it or make it work

File: Service/src/routes/league.py [hunk 10]
```diff
@@ -14,10 +14,10 @@
 @league_api.route('/api/league', methods=['POST'])
 def post_league():
    body = request.get_json()
-   try:
-      result = LeagueSchema().load(body)
-   except ValidationError as err:
-      return make_response(json.dumps(err.messages), 400)
+   # try:
```
Comment (ndara) [inline Service/src/routes/league.py]: again, no commented code, remove it if it's unused

beauty of Git/Github is that history is preserved

File: Service/src/routes/transaction.py [hunk 11]
```diff
@@ -90,7 +90,7 @@ def sell(sharePrice, requestBody):
    g.cursor.execute("INSERT INTO Transaction" +
       "(sharePrice, shareCount, action, portfolioId, ticker, leagueId) " +
       "VALUES (%s, %s, %s, %s, %s, %s)",
-      [sharePrice, requestBody['shareCount'], 0, requestBody['portfolioId'], requestBody['ticker'], portfolio['leagueId']])
+      [sharePrice, requestBody['shareCount'], "SELL", requestBody['portfolioId'], requestBody['ticker'], portfolio['leagueId']])
```
Comment (ndara) [inline Service/src/routes/transaction.py:93]: why is this change necessary?

File: Service/src/routes/watchlist.py [hunk 12]
```diff
@@ -9,6 +9,7 @@
 @watchlist_api.route('/api/watchlist', methods=['POST'])
 def post_watchlist():
    body = request.get_json()
+   g.log.info(str(body))
```
Comment (ndara) [inline Service/src/routes/watchlist.py]: use .debug when debugging

and this is also a blocker for merge because this log shouldn't be needed

File: Service/src/util/db_connection.py [hunk 13]
```diff
@@ -2,7 +2,7 @@
 import os
```
Comment (ndara) [inline Service/src/util/db_connection.py:2]: See how file path is configured here...replicate it for your iextoken

**Review by ndara — CHANGES_REQUESTED** [review]
* Add a PR description next time with brief description of changes
* You should be committing way more than once --> commit for every unique change
* Identified several blockers for merge, fix before merging

File: Scripts/setup.sh (hunk 1 above)
Comment (rohithdara) [reply Scripts/setup.sh:96]: Added back, was a mistake in debugging

File: Service/src/app.py (hunk 2 above)
Comment (rohithdara) [reply Service/src/app.py]: Removed

File: Service/src/app.py (hunk 3 above)
Comment (rohithdara) [reply Service/src/app.py:45]: The endpoints for every route didn't have versioning. It was either adding versions to each of them or removing it once at the app.py

File: Service/src/routes/charts.py (hunk 4 above)
Comment (rohithdara) [reply Service/src/routes/charts.py:17]: Seemed like it

File: Service/src/app.py (hunk 3 above)
Comment (ndara) [reply Service/src/app.py:45]: ok - this is resolved

File: Service/src/routes/league.py (hunk 10 above)
Comment (rohithdara) [reply Service/src/routes/league.py]: Removed

File: Service/src/routes/transaction.py (hunk 11 above)
Comment (rohithdara) [reply Service/src/routes/transaction.py:93]: Clarity, why call the action "0" instead of "SELL"

File: Service/src/util/db_connection.py (hunk 13 above)
Comment (rohithdara) [reply Service/src/util/db_connection.py:2]: I tried importing this function from this file and it would call the global var from the wrong config file. I'll just rewrite the function in the charts.py

File: Service/src/routes/charts.py (hunk 5 above)
Comment (rohithdara) [reply Service/src/routes/charts.py]: Fixed lol

File: Service/src/routes/charts.py (hunk 7 above)
Comment (rohithdara) [reply Service/src/routes/charts.py]: Not necessary anymore, so it is removed regardless

File: Service/src/routes/charts.py (hunk 6 above)
Comment (rohithdara) [reply Service/src/routes/charts.py]: Removed

File: Service/src/routes/charts.py (hunk 8 above)
Comment (rohithdara) [reply Service/src/routes/charts.py]: Removed

File: Service/src/routes/charts.py (hunk 9 above)
Comment (rohithdara) [reply Service/src/routes/charts.py]: Removed, and fixed the code that was leading to that exception... I had to filter out the improper output items from IEX itself

File: Service/src/routes/watchlist.py (hunk 12 above)
Comment (rohithdara) [reply Service/src/routes/watchlist.py]: Removed

File: Service/src/util/db_connection.py (hunk 13 above)
Comment (rohithdara) [reply Service/src/util/db_connection.py:2]: Followed the same model and fixed it

File: Service/src/routes/charts.py (hunk 7 above)
Comment (ndara) [reply Service/src/routes/charts.py]: why are the changes still here then?

File: Service/src/routes/transaction.py (hunk 11 above)
Comment (ndara) [reply Service/src/routes/transaction.py:93]: ok

Comment (ndara) [reply Service/src/routes/transaction.py:93]: keep in mind there are several ways to sell in a stock market, at present we only support "market sell" so "SELL" is definitely ambigious

that's probably why we did the numeric before

**Review by ndara — APPROVED** [review]
left comments but no merge blockers --> approved

File: Service/src/routes/charts.py (hunk 7 above)
Comment (rohithdara) [reply Service/src/routes/charts.py]: I deleted everything but the declaration, mistake
